app/models/time_limit.rb

```diff
@@ -0,0 +1,4 @@
+class TimeLimit < ApplicationRecord
```

Comment:
As discussed, I'm a little on the fence about having this in the database. Adding new rules in the database will require a PR anyway, so we don't win anything in terms of deployment speed. Have you considered to hard code the values in a way that doesn't require table, but keeps the nice extensibility of `TimeLimitCalculator`?

My preference is that the app code works with an empty database. Since `seeds.rb` doesn't run automatically, does this mean the app will error if we merge this PR as it is currently?

Reply:
It doesn't error with an empty DB - it just doesn't ever reject by default. To make it work with an empty database we'd need a hard-coded default. If we are going to do that then perhaps we should bring the whole config into code. The interface to `TimeLimitCalculator` can stay as it is even if we put the config into a constant collection of rules, e.g.:

```
  TimeLimit = Struct.new(:from_date, :to_date, :limit)
  RULES = {
    reject_by_default: [
      TimeLimit.new(nil, nil, 40),
      TimeLimit.new(Date.new(2020, 8, 1), Date.new(2020, 9, 30), 30),
      TimeLimit.new(Date.new(2020, 10, 1), Date.new(2020, 12, 31), 20),
    ],
  }.freeze
```

I'm on the fence too, happy to be pushed either way!

Having it in the database makes more sense if we think we are likely to have an 'admin' interface to tweak these rules but you could argue YAGNI.

Reply:
👍 Your hard coded config looks good, I'd be happy with that.

Reply:
I also like that approach!

Reply:
:+1: I've removed the DB table/model and added the hard-coded config.